packages/datadog-instrumentations/src/knex.js

```diff
@@ -18,3 +22,68 @@ function patch (file) {
     return Builder
   })
 }
+
+addHook({
+  name: 'knex',
+  versions: ['>=2'],
+  file: 'lib/knex-builder/Knex.js'
+}, Knex => {
+  shimmer.wrap(Knex.Client.prototype, 'raw', raw => function () {
+    if (!startRawQueryCh.hasSubscribers) {
+      return raw.apply(this, arguments)
+    }
+
+    const sql = arguments[0]
+
+    // Skip query done by Knex to get the value used for undefined
+    if (sql === 'DEFAULT') {
+      return raw.apply(this, arguments)
+    }
+
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+
+    function onFinish () {
+      asyncResource.bind(function () {
```

Comment:
Is this necessary? Each point that you are calling `onFinish` you are already in `asyncResource.runAsyncScope` (L68 and L78)